craft_providers/bases/buildd.py

```diff
@@ -0,0 +1,522 @@
+                capture_output=True,
+                check=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+
+            executor.execute_run(
+                ["systemctl", "restart", "systemd-networkd"],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to setup systemd-networkd.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def _setup_resolved(self, *, executor: Executor, deadline: Optional[float]) -> None:
+        """Configure system-resolved to manage resolve.conf.
+
+        :param executor: Executor for target container.
+        :param timeout_secs: Timeout in seconds.
+        """
+        try:
+            executor.execute_run(
+                [
+                    "ln",
+                    "-sf",
+                    "/run/systemd/resolve/resolv.conf",
+                    "/etc/resolv.conf",
+                ],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+
+            executor.execute_run(
+                ["systemctl", "enable", "systemd-resolved"],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+
+            executor.execute_run(
+                ["systemctl", "restart", "systemd-resolved"],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+        except subprocess.CalledProcessError as error:
+            raise BaseConfigurationError(
+                brief="Failed to setup systemd-resolved.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def _setup_snapd(
+        self, *, executor: Executor, deadline: Optional[float] = None
+    ) -> None:
+        """Install snapd and dependencies and wait until ready.
+
+        :param executor: Executor for target container.
+        :param deadline: Optional time.time() deadline.
+        """
+        try:
+            # TODO: Is udev required for LXD? it is not for Multipass.
+            executor.execute_run(
+                [
+                    "apt-get",
+                    "install",
+                    "-y",
+                    "fuse",
+                    "udev",
+                ],
+                check=True,
+                capture_output=True,
+                env=self.command_environment,
+            )
+            _check_deadline(deadline)
+
+            executor.execute_run(
```

Comment:
Since all these calls are the same except for the command being executed, perhaps we could iterate through a sequence of commands?

Reply:
They could, but I'd rather just spend a few extra lines and have the flexibility if I need to change a parameter or handling for just one call later.